### DISCUSSION TITLE
404 on bin/vite dev

### DISCUSSION THREAD

**Discussion by notturingtested** [ElMassimo/vite_ruby#131] [Q&A] [answered]
Hello everyone! I am new to ruby , vue and vite. I had a small project in ruby with vue that I am trying to get working in vite. I had it all working separated, but when I put it all together I can't get my vue code to serve. I created this repository to see if anyone could help me, you can find it here: https://github.com/notturingtested/notabee. if you run docker-compose up at the root of the project, the application will start, but once it is all booted, if you go to localhost:3036 you just get a 404. Could anyone help me trouble shoot my configuration? Thank you!

**Comment by jitingcn** [comment]
You might need to add `server: { fs: { strict: false } },` to `vite.config.ts`
I got an error after updrade to vite 2.6.5 recently.
```
Unrestricted file system access to "/entrypoints/application.js"
For security concerns, accessing files outside of serving allow list will be restricted by default in the future version of Vite. Refer to https://vitejs.dev/config/#server-fs-allow for more details.
```

**Comment by ElMassimo** [comment, marked as answer]
@notturingtested You should not visit the Vite dev server, in Vite Ruby projects it's only used to serve assets. Visit the Rails server instead.

A few things I noticed:

- Your entrypoint is not mounting the Vue application, you do that in app/frontend/main.js, but it's not imported. Move the content to entrypoints/application.js or import `~/main` from there.
- The application.html.erb layout is still using the webpacker tags

**Reply by jitingcn** [reply]
> I got an error after updrade to vite 2.6.5 recently.
> 
> ```
> Unrestricted file system access to "/entrypoints/application.js"
> For security concerns, accessing files outside of serving allow list will be restricted by default in the future version of Vite. Refer to https://vitejs.dev/config/#server-fs-allow for more details.
> ```

So I might have a different issue?

**Reply by ElMassimo** [reply]
Which version of vite-plugin-ruby is installed?

In recent releases, [`fs.allow` is set](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/src/index.ts#L37) by the plugin.

**Reply by jitingcn** [reply]
Seem like I forgot to update node_module, now it worked. Thanks!

**Reply by ElMassimo** [reply]
Cool! Recent versions of Vite Ruby include a `bin/vite upgrade` command that you can use to upgrade the gems, the plugin, and vite all at once.

**Reply by notturingtested** [reply]
I copied everything from main.js to application.js, but I still cant see App.vue, just the "Yay! You’re on Rails!" page.

**Reply by ElMassimo** [reply]
That's because the app [doesn't have a `/` route](https://github.com/notturingtested/notabee/blob/main/config/routes.rb#L1).

Try visiting `/notes`, and notice there aren't any views for the notes controller.
